src/main/java/br/fatec/bd4/repository/RegistroRepository.java

```diff
@@ -2,22 +2,34 @@
 
 import br.fatec.bd4.entity.Registro;
 
+import br.fatec.bd4.web.dto.MaxMinDTO;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
+import java.util.Optional;
+
 @Repository
 public interface RegistroRepository extends JpaRepository<Registro, Long> {
 
+
+    @Query(
+        "SELECT new br.fatec.bd4.web.dto.MaxMinDTO("+
+        "MAX(R.local.latitude), MAX(R.local.longitude), MIN(R.local.latitude), MIN(R.local.longitude)) " +
+        "FROM Registro R WHERE R.usuario.id = :idUsuario"
+    )
+    MaxMinDTO findMaxRegistro(@Param("idUsuario") Long idUsuario);
+
     @Query(
-        "SELECT R FROM Registro R WHERE R.dataHora BETWEEN TO_TIMESTAMP(:startDate, 'YYYY-MM-DD HH24:MI:SS') AND TO_TIMESTAMP(:endDate, 'YYYY-MM-DD HH24:MI:SS') " +
-        "AND R.usuario.id = :idUsuario " +
-        "ORDER BY R.dataHora ASC"
+            value = "SELECT * FROM registro R WHERE R.data_hora BETWEEN TO_TIMESTAMP(:startDate, 'YYYY-MM-DD HH24:MI:SS') AND TO_TIMESTAMP(:endDate, 'YYYY-MM-DD HH24:MI:SS') " +
+                    "AND R.usuario_id = :idUsuario " +
+                    "ORDER BY R.data_hora ASC",
+            nativeQuery = true
```

Comment:
prefira utilizar JPql, query nativa não passa pela validação do spring

Reply:
You're right, nativeQuery removed from RegistroRepository.